functions/index.js

```diff
@@ -20,47 +23,222 @@ function getHelpType(helpIdx) {
     return "unknown";
   }
 }
-
+ 
+// puts a clap in the database
 exports.addClap = functions.https.onRequest( async(req, res) => {
   res.set('Cache-Control', 'public, max-age=300, s-maxage=600');
-  const country_code = req.headers["x-appengine-country"] || 'ZZ';
-  var clapsRef = db.ref('claps');
-  var newClap = clapsRef.push();
-  clap_data = {
-    country: country_code,
-    createdAt: Date.now(),
-  };
-  newClap.set(clap_data);
-  res.status(200).json(clap_data);
+  return cors(req, res, async () => {
+    resetDayCount();
+    const country_code = req.headers["x-appengine-country"];
+    //const country_code = "CO";
+    var countryRef = db.ref('claps/'+country_code);
+    var todayRef = db.ref('claps/today');
+    var totalRef = db.ref('claps/total');
+
+    var returnData;
+    var totalCount = -1;
+    var todayCount = -1;
+    var countryCount = -1;
+
+    //increases total claps
+    await totalRef.once("value", function(snapshot) {
+      totalCount = snapshot.val().count;
```

Comment:
Every call dependent on the database being populated either fails or returns an unexpected result form the api (`null` instead of a json). They should silently fail on non-existent snapshots by either returning 0 when asking for the count, or returning an empty json like in the case of the `getXXFoundation` calls. The cleanest solution I can think of is creating a function that receives a snapshot and returns the expected value when non-existent (0 for count, {} for other calls).

Even though we ideally won't deal with this situation on production, every time we start our local emulation we work with an empty database, so this is a relevant change.